attending/downloader.py

```diff
@@ -0,0 +1,38 @@
+import urllib.request
+from pathlib import Path
+from zipfile import ZipFile
+
+from .Doc import DocLocation
+
+
+def extract_zip(doc_location: DocLocation, file: Path):
+    ZipFile(file).extractall(path=doc_location.full_path())
+    file.unlink()
+
+
+def write_to_file(url, doc_location: DocLocation):
+    with urllib.request.urlopen(url) as connection:
+        if 400 > connection.status >= 300:
+            write_to_file(connection.getheader("Location"), doc_location)
+        elif connection.status == 200:
+            file_extension = _EXTENSION_MAPPING.get(connection.getheader('Content-Type'), 'txt')
+            target = doc_location.full_path() / Path(f"{doc_location.name}.{file_extension}")
+            with open(target, "wb") as f:
+                for bytes in range(0, int(connection.getheader("Content-Length")), 1024):
+                    f.write(connection.read(bytes))
```

Comment:
I don't think this is correct - do you really mean to read the connection in chunks that increase by 1024 bytes, starting with 0 bytes?

Reply:
reading 0 bytes would be very efficient.